### PR TITLE
Prevent use of older clients with already-upgraded databases

### PR DESCRIPTION
**Pull request by erikjv**
Fixes: #10964

### CODE REVIEW

File: src/common/syncjournaldb.cpp [hunk 1]
```diff
@@ -192,6 +192,27 @@ bool SyncJournalDb::maybeMigrateDb(const QString &localPath, const QString &abso
     return true;
 }
 
+bool SyncJournalDb::checkVersion(const QString &dbFilePath)
```
Comment (TheOneRing) [inline src/common/syncjournaldb.cpp:195]: Why is naming things so hard, I suggested checkVersion but I'm not happy 🙈 
Maybe checkDowngrade? In which case we would need to invert the condition.

**Review by TheOneRing — CHANGES_REQUESTED** [review]

File: src/gui/main.cpp [hunk 2]
```diff
@@ -338,6 +344,17 @@ int main(int argc, char **argv)
         platform->setApplication(&app);
 
         auto ocApp = new OCC::Application(platform.get(), options.debugMode, &app);
+
+        if (!FolderMan::instance()->setupFolders().has_value()) {
```
[the remark this answers is not in the document]
Comment (erikjv) [reply src/gui/main.cpp:348]: it does, see the comment I added

File: src/common/syncjournaldb.cpp (hunk 1 above)
Comment (erikjv) [reply src/common/syncjournaldb.cpp:195]: Done, but is the version check correct? Meaning, do we only want to check the major version?

Comment (TheOneRing) [reply src/common/syncjournaldb.cpp]: As minor version changes are generally mean that there where no breaking changes I'd hope comparing the major version is sufficient.

**Review by TheOneRing — APPROVED** [review]

**Pull request merged**